react/hooks/useFetchMore.js

```diff
@@ -185,6 +197,141 @@ export const useFetchMore = (props) => {
     isFirstRender.current = false
   }, [maxItemsPerPage, query, map, orderBy, priceRange])
 
+  //PAGINATION
```

Comment:
I don't think the way you implemented the fetch will help you with the SEO problem.

The buttons you created will have the same behavior as the "See More" button. You are just fetching it using javascript. I think Google recommends to use the history API

I would recommend checking the [render-runtime](https://github.com/vtex-apps/render-runtime) project and looking for the `Link` component and the `navigate` function. Check if they use the history API